devito/finite_differences/derivative.py

```diff
@@ -497,3 +580,141 @@ def _eval_fd(self, expr, **kwargs):
             res = res.xreplace(e)
 
         return res
+
+    def _eval_expand_nest(self, **hints):
+        """
+        Expands nested derivatives
+        `Derivative(Derivative(f(x), (x, b)), (x, a))
+            --> Derivative(f(x), (x, a+b))`
+        `Derivative(Derivative(f(x), (y, b)), (x, a))
+            --> Derivative(f(x), (x, a), (y, b))`
+        Note that this is not always a valid expansion depending on the kwargs
+        used to construct the derivative.
+        """
+        if not isinstance(self.expr, self.__class__):
+            return self
+
+        # This is necessary as tools.abc.Reconstructable._rebuild will copy
+        # all kwargs from the self object. Need to enssure that the nest is not
+        # actually expanded if derivatives are incompatible.
+        # The nested derivative is evaluated by:
+        # 1. Chaining together the variables with which to differentiate wrt
+        new_expr = self.expr.args[0]
+        new_dims = [
+            (d, ii)
+            for d, ii in zip(
+                chain(self.dims, self.expr.dims),
+                chain(self.deriv_order, self.expr.deriv_order),
+                strict=True
```

Comment:
?

Reply:
the `strict` yes

Reply:
This ensures that the two iterables are of the same length

https://docs.python.org/3/library/functions.html#zip

New in 3.10

Reply:
I think this is resolved, open again if you aren't convinced